x/stakeibc/migrations/v3/convert_test.go

```diff
@@ -0,0 +1,188 @@
+package v3
+
+import (
+	"testing"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/stretchr/testify/require"
+
+	oldstakeibctypes "github.com/Stride-Labs/stride/v13/x/stakeibc/migrations/v3/types"
+	newstakeibctypes "github.com/Stride-Labs/stride/v13/x/stakeibc/types"
+)
+
+func TestConvertToNewValidator(t *testing.T) {
+	name := "name"
+	address := "address"
+	weight := uint64(3)
+	delegation := sdk.NewInt(4)
+	sharesToTokensRate := sdk.NewDec(5)
+	tvl := sdk.NewInt(1_000_000)
+	slashCheckpoint := sdk.NewInt(10_000) // 1% of TVL
+
+	// First convert a validator with no exchange rate
+	// It should get filled in with the default
+	oldValidator := oldstakeibctypes.Validator{
+		Name:          name,
+		Address:       address,
+		DelegationAmt: delegation,
+		Weight:        weight,
+	}
+	expectedNewValidator := newstakeibctypes.Validator{
+		Name:                        name,
+		Address:                     address,
+		Weight:                      weight,
+		Delegation:                  delegation,
+		SlashQueryProgressTracker:   sdkmath.ZeroInt(),
+		SlashQueryCheckpoint:        slashCheckpoint,
+		DelegationChangesInProgress: 0,
+		SharesToTokensRate:          DefaultExchangeRate,
+		SlashQueryInProgress:        false,
+	}
+
+	actualNewValidator := convertToNewValidator(oldValidator, tvl)
+	require.Equal(t, expectedNewValidator, actualNewValidator)
+
+	// Then add an exchange rate and convert again
+	oldValidator.InternalExchangeRate = &oldstakeibctypes.ValidatorExchangeRate{
+		InternalTokensToSharesRate: sharesToTokensRate,
+		EpochNumber:                1,
+	}
+	expectedNewValidator.SharesToTokensRate = sharesToTokensRate
+
+	actualNewValidator = convertToNewValidator(oldValidator, tvl)
+	require.Equal(t, expectedNewValidator, actualNewValidator)
+}
+
+func TestConvertToNewHostZone(t *testing.T) {
+	chainId := "chain"
+	connectionId := "connection"
+	bechPrefix := "bech"
+	channelId := "channel"
+	ibcDenom := "ibc"
+	hostDenom := "host"
+
+	depositAddress := "address"
+	withdrawalAddress := "withdrawal"
+	feeAddress := "fee"
+	delegationAddress := "delegation"
+	redemptionAddress := "redemption"
+
+	redemptionRate := sdk.NewDec(1)
+	lastRedemptionRate := sdk.NewDec(2)
+	minRedemptionRate := sdk.MustNewDecFromStr("0.95")
+	maxRedemptionRate := sdk.MustNewDecFromStr("1.25")
+	unbondingFrequency := uint64(4)
+	unbondingPeriod := uint64(21)
+
+	halted := true
+
+	valAddress := "val"
+	valDelegation := sdk.NewInt(5)
+	valWeight := uint64(6)
+	totalDelegations := sdk.NewInt(1_000_000)
+	slashCheckpoint := sdk.NewInt(10_000) // 1% of TVL
+	sharesToTokensRate := sdk.MustNewDecFromStr("0.99")
```

Comment:
Might make sense to add expected test values for a `InternalTokensToSharesRate`& `sharesToTokensRate` conversion.

Reply:
commented above - the original name was misleading, there's no conversion